src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,48 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] matrixDiagonal = new int [matrix.length];
+
+        for (int i = 0; i < matrix.length; i++) {
+            matrixDiagonal[i] = matrix[i][i];
+        }
+        return matrixDiagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int [matrix.length];
+
+        for (int i = 0; i < matrix.length; i++) {
+            counterDiagonal[i] = matrix[i][matrix[i].length - i - 1];
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = matrix[0][0];
+
+        for (int[] rowMatrix:matrix) {
```

Comment:
```suggestion
        for (int[] rowMatrix : matrix) {
```
don't forget about spaces in the code